features/305-microservice/microservice.md

```diff
@@ -14,16 +16,47 @@ deployments:
     port: 8888
     context: ./backend/service
     dockerfile: Dockerfile
+    command: "<start server command>"
 ```
 
-When the user run `skycli app deploy my-awesome-service`, the following steps are taken:
+*`dockerfile`, `command` are optional.*
+
+When the user run `skycli app deploy`, the following steps are taken:
 
 1. Assert `./backend/service` to be a directory.
 1. If `dockerfile` is specified, it is relative to `context`. Otherwise, the default value is `context/Dockerfile`.
 1. Assert the Dockerfile to be a file.
 1. Look for `<context>/.dockerignore`. If it is not present, the entire context is archived. Otherwise, use the same mechanism as `.dockerignore` to ignore files.
 1. Upload the archive and save as an artifact.
-1. Follow the existing deployment flow.
+1. Follow the existing docker deployment flow.
+1. Command is optional. If it is specified, it will override the default command.
+
+### Microservice with preconfigured environment (e.g. nodejs)
+
+```yaml
+deployments:
+  my-awesome-service:
+    type: http-service
+    path: /api/
+    port: 3000
+    context: ./backend/nodejs
+    runtime: nodejs
+    runtime_version: "10"
+    command: "npm start"
+```
+
+*`command`, `runtime_version` are optional.*
+
+When the user run `skycli app deploy`, the following steps are taken:
+
+1. Assert `./backend/service` to be a directory.
+1. If `runtime` is specified, `skycli` will request the controller with `runtime` and `runtime_version` to download the template in `tarball` format.
+1. Unzip the template and merge to the user code. Look for `<context>/.dockerignore`. If it is not present, the entire context is archived. Otherwise, use the same mechanism as `.dockerignore` to ignore files.
```

Comment:
I think the merge process need to be elaborate more. For example, what if the both the template and user code has `.dockerignore`?